types/catbox/index.d.ts

```diff
@@ -23,8 +23,8 @@ export interface CallBackWithResult<T> {
  *  * partition - the partition name used to isolate the cached results across multiple clients. The partition name is used as the MongoDB database name, the Riak bucket, or as a key prefix in Redis and Memcached. To share the cache across multiple clients, use the same partition name.
  * @see {@link https://github.com/hapijs/catbox#client}
  */
-export interface Client extends ClientApi {
-    new(engine: EnginePrototypeOrObject, options: ClientOptions): Client;
+export class Client extends ClientApi {
```

Comment:
Change this to `implements ClientApi`, rather than `extends`, then implement the methods appropriately (likely just copying definitions from the interface to it if required).